# Copyright 2007 Sun Microsystems, Inc.  All rights reserved.
#
# Permission is hereby granted, free of charge, to any person obtaining a
# copy of this software and associated documentation files (the
# "Software"), to deal in the Software without restriction, including
# without limitation the rights to use, copy, modify, merge, publish,
# distribute, and/or sell copies of the Software, and to permit persons
# to whom the Software is furnished to do so, provided that the above
# copyright notice(s) and this permission notice appear in all copies of
# the Software and that both the above copyright notice(s) and this
# permission notice appear in supporting documentation.
#
# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS
# OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
# MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT
# OF THIRD PARTY RIGHTS. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR
# HOLDERS INCLUDED IN THIS NOTICE BE LIABLE FOR ANY CLAIM, OR ANY SPECIAL
# INDIRECT OR CONSEQUENTIAL DAMAGES, OR ANY DAMAGES WHATSOEVER RESULTING
# FROM LOSS OF USE, DATA OR PROFITS, WHETHER IN AN ACTION OF CONTRACT,
# NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF OR IN CONNECTION
# WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
#
# Except as contained in this notice, the name of a copyright holder
# shall not be used in advertising or otherwise to promote the sale, use
# or other dealings in this Software without prior written authorization
# of the copyright holder.
# 

Preserve compatibility with old xorg.conf files that have "keyboard" or
"Keyboard" drivers listed.   (Sun bug 6560332)

https://bugs.freedesktop.org/show_bug.cgi?id=11301
Debian bug#428794

Index: hw/xfree86/common/xf86Init.c
===================================================================
--- hw/xfree86/common/xf86Init.c.orig	2007-07-14 12:14:29.000000000 +0200
+++ hw/xfree86/common/xf86Init.c	2007-07-14 12:15:21.000000000 +0200
@@ -1032,6 +1032,14 @@
 
 		continue;
 	    }
+#else
+	    /* Replace obsolete keyboard driver with kbd */
+    	    if (!xf86NameCmp(pDev->driver, "keyboard")) {
+		xf86MsgVerb(X_WARNING, 0, "*** WARNING the legacy keyboard driver \"%s\" is deprecated\n", pDev->driver);
+		xf86MsgVerb(X_WARNING, 0, "*** Using the new \"kbd\" driver for \"%s\".\n",
+			pDev->identifier);
+		strcpy(pDev->driver, "kbd");
+	    }
 #endif
 
 	    if ((pDrv = MatchInput(pDev)) == NULL) {
@@ -1951,6 +1959,11 @@
 	if (name == NULL || *name == '\0')
 	    continue;
 
+	/* Replace obsolete keyboard driver with kbd */
+	if (!xf86NameCmp(name, "keyboard")) {
+	    strcpy(name, "kbd");
+	}
+	
 	if (optlist)
 	    opt = optlist[i];
 	else
